orchestrator/gravity_utils/src/prices.rs

```diff
@@ -21,23 +24,94 @@ pub async fn get_weth_price_with_retries(
         return Ok(0u8.into());
```

Comment:
This is the function used in should_relay_batch(). I have modified it to work in 2 phases:
* First the direct token -> WETH pools (all fee levels) are queried on Uniswap v3, and Uniswap v2 is also queried.
  * If both return results, v3 is preferred
* Then the Uniswap v3 stablecoin intermediate path is tested: token -> {USDC, USDT, DAI} (all fee levels) -> WETH